### PR TITLE
Utkarsha/refinement api token accounts

### PR DESCRIPTION
**Pull request by utkarsha-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/account/src/Components/api-token/api-token.tsx [hunk 1]
```diff
@@ -231,45 +216,35 @@ const ApiToken = ({ footer_ref, is_app_settings, overlay_ref, setIsOverlayShown
                                                 <div className='da-api-token__checkbox-wrapper'>
                                                     <ApiTokenCard
                                                         name='read'
-                                                        value={values.read}
-                                                        setFieldValue={setFieldValue}
                                                         display_name={localize('Read')}
                                                         description={localize(
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/api-token.tsx]: can we make this to Localize 
This description is rendered as children of Text component

Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/api-token.tsx]: Should I do it for description only or `InlineNoteWithIcon` component as well?

File: packages/translations/src/translations/ar.json [hunk 2]
```diff
@@ -540,7 +540,7 @@
     "619407328": "هل تريد بالتأكيد إلغاء الارتباط من {{identifier_title}}؟",
     "623192233": "يرجى إكمال <0>اختبار الملاءمة</0> للوصول إلى الكاشير / أمين الصندوق الخاص بك.",
     "623542160": "مصفوفة المتوسط المتحرك الأسي (EMAA)",
-    "624668261": "You’ve just stopped the bot. Any open contracts can be viewed on the <0>Reports</0> page.",
+    "624668261": "لقد أوقفت الروبوت للتو. يمكن عرض أي عقود مفتوحة على صفحة <0>التقارير</0>.",
```
Comment (amina-deriv) [inline packages/translations/src/translations/ar.json]: why do we have these translation changes?

can you check this

File: packages/account/src/Components/api-token/api-token.tsx [hunk 3]
```diff
@@ -231,54 +216,44 @@ const ApiToken = ({ footer_ref, is_app_settings, overlay_ref, setIsOverlayShown
...
                                                         name='trading_information'
-                                                        value={values.trading_information}
                                                         display_name={localize('Trading information')}
-                                                        setFieldValue={setFieldValue}
-                                                        description={localize(
-                                                            'This scope will allow third-party apps to view your trading history.'
-                                                        )}
+                                                        description={
+                                                            <Localize i18n_default_text='This scope will allow third-party apps to view your trading history.' />
+                                                        }
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/api-token.tsx]: can we create an array with name,description and display name and iterate through it

File: packages/account/src/Components/api-token/api-token-card.tsx [hunk 4]
```diff
@@ -3,38 +3,26 @@ import { Field, FieldProps } from 'formik';
 import { CompositeCheckbox } from '@deriv/components';
 
 type TApiTokenCard = {
-    description: string;
+    description: JSX.Element;
```
Comment (shahzaib-deriv) [inline packages/account/src/Components/api-token/api-token-card.tsx:6]:
```suggestion
    description: React.ReactNode;
```

File: packages/components/src/components/composite-checkbox/composite-checkbox.tsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ type TCompositeCheckbox = {
     className?: string;
     label: string;
     id?: string;
-    description: string;
+    description: JSX.Element;
```
Comment (shahzaib-deriv) [inline packages/components/src/components/composite-checkbox/composite-checkbox.tsx:14]:
```suggestion
    description: React.ReactNode;
```

File: packages/core/src/Stores/ui-store.js [hunk 6]
```diff
@@ -491,6 +492,10 @@ export default class UIStore extends BaseStore {
         this.is_close_uk_account_modal_visible = is_open;
     }
 
+    get is_desktop() {
+        return this.screen_width > MAX_MOBILE_WIDTH;
+    }
+
```
Comment (shahzaib-deriv) [inline packages/core/src/Stores/ui-store.js:498]: Do we really need this flag? As the opposite of `is_mobile` can be desktop.
You can skip this comment as well

File: packages/account/src/Constants/api-token-card-details.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        name: 'payments',
+        display_name: localize('Payments'),
+        description: (
+            <Localize i18n_default_text='This scope will allow third-party apps to withdraw to payment agents and make inter-account transfers for you.' />
+        ),
+    },
+    {
+        name: 'trading_information',
+        display_name: localize('Trading information'),
```
Comment (amina-deriv) [inline packages/account/src/Constants/api-token-card-details.tsx]: CompositeCheckbox is only used in api-token-card. CompositeCheckbox uses checkbox inside which accepts ReactElement.
 can we make this display name to Localize component and change the typeof TCompositeCheckbox accordingly.

File: packages/account/src/Components/api-token/api-token-card.tsx (hunk 4 above)
Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/api-token-card.tsx:6]: description can't be null or undefined ...I think JSX.Element would be a better choice?

File: packages/account/src/Constants/api-token-card-details.tsx (hunk 7 above)
Comment (utkarsha-deriv) [reply packages/account/src/Constants/api-token-card-details.tsx]: Sure thanks

File: packages/account/src/Components/api-token/_tests_/api-token-article.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import ApiTokenArticle from '../api-token-article';
+
+it('should render ApiTokenArticle', () => {
+    render(<ApiTokenArticle />);
+    expect(screen.getByText('API token'));
+    expect(
+        screen.getByText(
+            `To access your mobile apps and other third-party apps, you'll first need to generate an API token.`
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-article.spec.tsx]: For test cases, it is always better to use regular expression for text match

File: packages/account/src/Components/api-token/_tests_/api-token-card.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ApiTokenCard from '../api-token-card';
+import { Formik, Form } from 'formik';
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-card.spec.tsx]: lets reorder these imports based on import rule https://github.com/binary-com/deriv-app/blob/master/docs/JavaScript/README.md#import-rules

File: packages/account/src/Components/api-token/_tests_/api-token-card.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        setFieldValue: jest.fn(),
+    };
+
+    it('should render ApiTokenCard', () => {
+        render(
+            <Formik initialValues={{ [mock_props.name]: mock_props.value }} onSubmit={jest.fn()}>
+                <Form>
+                    <ApiTokenCard {...mock_props} />
+                </Form>
+            </Formik>
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-card.spec.tsx]: can we make this render to a common function

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+    const mock_token = {
+        token: {
+            display_name: 'Token 1',
+            last_used: '12/31/2022',
+            scopes: ['read', 'trade'],
+            token: '1234567',
+        },
+    };
+
+    const renderAPiDeleteButton = () => {
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: renderAPIDeleteButton

File: packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    });
+
+    it('should display Info Message when user clicks on OK', async () => {
+        render(<ApiTokenClipboard {...mock_props} />);
+        const copy_icon = screen.getByTestId('dt_copy_token_icon');
+        userEvent.click(copy_icon);
+        expect(screen.getByRole('button', { name: 'OK' })).toBeInTheDocument();
+        const ok_button = screen.getByRole('button', { name: 'OK' });
+        userEvent.click(ok_button);
+        await waitFor(() => {
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx]: do we need to await here?

File: packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    it('should render ApiTokenClipboard with the copy icon', () => {
+        render(<ApiTokenClipboard {...mock_props} />);
+        expect(screen.getByTestId('dt_copy_token_icon')).toBeInTheDocument();
+    });
+
+    it('should display "Copy this token" message when mouse enters', () => {
+        render(<ApiTokenClipboard {...mock_props} />);
+        const copy_icon = screen.getByTestId('dt_copy_token_icon');
+        userEvent.hover(copy_icon);
+        expect(screen.getByText('Info Message')).toBeInTheDocument();
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx]: description says 'should display "Copy this token"' but we are checking 'Info Message' here.

File: packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    });
+
+    it('should remove "Copy this token" message when mouse leaves', () => {
+        render(<ApiTokenClipboard {...mock_props} />);
+        const copy_icon = screen.getByTestId('dt_copy_token_icon');
+        userEvent.hover(copy_icon);
+        expect(screen.getByText('Info Message')).toBeInTheDocument();
+        userEvent.unhover(copy_icon);
+        expect(screen.queryByText('Info Message')).not.toBeInTheDocument();
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx]: here too

File: packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+        render(<ApiTokenClipboard {...mock_props} />);
+        const copy_icon = screen.getByTestId('dt_copy_token_icon');
+        userEvent.click(copy_icon);
+        expect(screen.getByRole('button', { name: 'OK' })).toBeInTheDocument();
+        const ok_button = screen.getByRole('button', { name: 'OK' });
+        userEvent.click(ok_button);
+        await waitFor(() => {
+            expect(screen.queryByText('Add accounts')).not.toBeInTheDocument();
+        });
+        expect(screen.getByText('Info Message')).toBeInTheDocument();
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx]: This is the only assertion we are checking additionally from previous test case. Maybe we can club them.

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    it('should display Delete this token when mouse enter', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+    });
+
+    it('should not Delete this token when mouse leave', () => {
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: As per test case, we are checking 'should not display Delete this token message when mouse leave'. can you check the test case here

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    it('should close the modal when clicked on Cancel', async () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        const cancel_button = screen.getByRole('button', { name: 'Cancel' });
+        userEvent.click(cancel_button);
+        await waitFor(() => expect(screen.queryByText('Delete token')).not.toBeInTheDocument());
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: can you check if this is working without await

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+        expect(screen.queryByText('Are you sure you want to delete this token?')).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Cancel' })).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Yes, delete' })).not.toBeInTheDocument();
+    });
+
+    it('should should trigger deleteToken when clicked on Yes, delete', async () => {
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: we are not using await anywhere in this test case. Do we need to wrap it async then?

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        const delete_token_button = screen.getByRole('button', { name: 'Yes, delete' });
+        act(() => {
```
Comment (amina-deriv) [inline packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: do we need act here. i think userEvent is already wrapped in act. Can u check this

File: packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx (hunk 12 above)
Comment (suisin-deriv) [reply packages/account/src/Components/api-token/_tests_/api-token-clipboard.spec.tsx]: Yes we need the await function here

File: packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx (hunk 17 above)
Comment (suisin-deriv) [reply packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx]: We need this await function here

**Review by amina-deriv — APPROVED** [review]

**Comment by amina-deriv** [comment]
Nicely done. LGTM 👌

**Review by matin-deriv — APPROVED** [review]
Nice one, LGTM 👍🏼

**Pull request merged**